frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -185,6 +214,82 @@ const MenuListItem = forwardRef<HTMLLIElement, MenuListItemProps>(
 function MainMenu(props: Props): ReactElement {
   const isServerDisconnected = !props.isServerConnected
 
+  const onClickDeployApp = useCallback((): void => {
+    const {
+      deployParams,
+      showDeployError,
+      closeDialog,
+      isDeployErrorModalOpen,
+    } = props
+
+    if (
+      (!deployParams ||
+        !deployParams.repository ||
+        !deployParams.branch ||
+        !deployParams.module) &&
+      !deployParams?.isHeadDetached
+    ) {
+      const dialog = NoRepositoryDetected()
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (deployParams?.isHeadDetached) {
+      const dialog = DetachedHead()
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (
+      deployParams?.module &&
+      deployParams?.untrackedFiles?.includes(deployParams.module)
+    ) {
+      const dialog = ModuleIsNotAdded(deployParams.module)
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (deployParams?.module && deployParams?.uncommittedFiles?.length) {
+      const dialog = UncommittedChanges(deployParams.module)
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (deployParams?.isAhead) {
+      const dialog = RepoIsAhead()
+
+      showDeployError(dialog.title, dialog.body, getDeployAppUrl(deployParams))
+
+      return
+    }
+
+    if (deployParams?.untrackedFiles?.length) {
+      const dialog = UntrackedFiles()
+
+      showDeployError(dialog.title, dialog.body, getDeployAppUrl(deployParams))
+
+      return
+    }
+
+    if (isDeployErrorModalOpen) closeDialog()
```

Comment:
I have to do because the state of the dialog is being handled on App, so in the case that you have a dialog , and then you solve the problem and then "Try Again" the dialog will close and use `getDeployAppUrl` to do the open.window with the correct coordinates

Reply:
I'll add a comment

Reply:
done